packages/react/src/TreeView/TreeView.module.css

```diff
@@ -0,0 +1,272 @@
+    /* stylelint-disable-next-line primer/spacing */
+    padding-top: calc((var(--min-item-height) - var(--custom-line-height, 1.3rem)) / 2);
+    /* stylelint-disable-next-line primer/spacing */
+    padding-bottom: calc((var(--min-item-height) - var(--custom-line-height, 1.3rem)) / 2);
+    line-height: var(--custom-line-height, var(--text-body-lineHeight-medium, 1.4285));
+    grid-area: content;
+    gap: var(--stack-gap-condensed);
+  }
+
+  .PRIVATE_TreeViewItemContentText {
+    flex: 1 1 auto;
+    width: 0;
+  }
+
+  &[data-truncate-text='true'] .PRIVATE_TreeViewItemContentText {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+  }
+
+  &[data-truncate-text='false'] .PRIVATE_TreeViewItemContentText {
+    word-break: break-word;
+  }
+
+  .PRIVATE_TreeViewItemVisual {
+    display: flex;
+
+    /* The visual icons should appear vertically centered for single-line items, but remain at the top for items that wrap
+    across more lines. */
+    height: var(--custom-line-height, 1.3rem);
+    color: var(--fgColor-muted);
+    align-items: center;
+  }
+
+  .PRIVATE_TreeViewItemLeadingAction {
+    display: flex;
+    color: var(--fgColor-muted);
+    grid-area: leadingAction;
+  }
+
+  .PRIVATE_TreeViewItemLevelLine {
+    width: 100%;
+    height: 100%;
+
+    /*
+     * On devices without hover, the nesting indicator lines
+     * appear at all times.
+     */
+    border-color: var(--borderColor-muted);
+    border-right: var(--borderWidth-thin) solid;
+  }
+
+  /*
+   * On devices with :hover support, the nesting indicator lines
+   * fade in when the user mouses over the entire component,
+   * or when there's focus inside the component. This makes
+   * sure the component remains simple when not in use.
+   */
+  @media (hover: hover) {
+    .PRIVATE_TreeViewItemLevelLine {
+      border-color: transparent;
+    }
+
+    &:hover .PRIVATE_TreeViewItemLevelLine,
+    &:focus-within .PRIVATE_TreeViewItemLevelLine {
+      border-color: var(--borderColor-muted);
+    }
+  }
+
+  .PRIVATE_TreeViewDirectoryIcon {
+    display: grid;
+    color: var(--treeViewItem-leadingVisual-iconColor-rest);
+  }
+
+  .PRIVATE_TreeViewVisuallyHidden {
+    position: absolute;
+    width: 1px;
+    height: 1px;
+    padding: 0;
+    /* stylelint-disable-next-line primer/spacing */
+    margin: -1px;
+    overflow: hidden;
+    clip: rect(0, 0, 0, 0);
+    white-space: nowrap;
+    border-width: 0;
+  }
+}
+
+/*
+ * Using "legacy" variable and hex code fallback as a workaround for dark high
+ * contrast mode, which currently does not support the expected color for
+ * `--treeViewItem-leadingVisual-iconColor-rest` variable.
+ */
+/* stylelint-disable-next-line selector-no-qualifying-type */
+body[data-dark-theme='dark_high_contrast'] .PRIVATE_TreeViewDirectoryIcon {
+  /* stylelint-disable-next-line primer/colors */
+  color: var(--color-tree-view-item-directory-fill, #f0f3f6);
+}
```

Comment:
See [this comment](https://github.com/primer/react/pull/5267#issuecomment-2474389109) or the [commit message](https://github.com/primer/react/pull/5267/commits/e97cf727641632e763b5e01966786316095040d2) for more context on this workaround.

Reply:
Hopefully you can remove this after pulling main! 🤞

Reply:
Thank you so much! All good now, workaround removed in https://github.com/primer/react/pull/5267/commits/fbe385d195e0a221c63b860731820a7ef4047424 🎉